[P2719] Add additional design choice for mandatory parameters (#11)
ldionne/wg21: p2719.md

> scope ([basic.scope.scope]{.sref .unnumbered}), otherwise the program is ill-formed.
:::

## Design choice: All implicit parameters are mandatory

**New since Wroclaw**

The proposal currently continues to consider optional implicit parameters to be optional. During
edge case testing we found that the non-universality of sized deallocation creates a hazard where
a less constrained sized deallocation may be selected over a more constrained unsized deallocation
function. We believe it is actually reasonable to update the language to make all currently optional
implicit parameters mandatory for type aware allocators.

> We believe step to take as in future any type oriented implicit parameters can be based on

Reviewer: ```suggestion
We believe this is a reasonable step to take, as in the future any type-oriented implicit parameters can be based on
```

> `type_traits` or similar functionality, and so reduce the exponential increase in new allocation and

Reviewer: ```suggestion
the `std::type_identity` parameter, which would reduce the exponential increase in new allocation and
```
